UOP1_Project/Assets/Scripts/SpawnSystem.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Linq;
+using UnityEditor.PackageManager;
+using UnityEngine;
+using UnityEngine.Events;
+
+[Serializable]
+public class PlayerSpawnEvent : UnityEvent<Transform> { }
+
+public class SpawnSystem : MonoBehaviour
+{
+    private static int requestedSpawnIndex = -1;
+
+    [Header("Settings")]
+    [SerializeField] private int defaultSpawnIndex = 0;
+
+    [Header("Project References")]
+#pragma warning disable 649
```

Comment:
Is this to avoid the nasty warning where unity says this value is never initialized? Could we disable this warning by default for the project instead? This can be done a csc.rsp file with the following:

-nowarn:0649

Reply:
I also don't think this is necessary.